ngraph/src/ngraph/pass/pass.hpp

```diff
@@ -79,14 +87,45 @@ class NGRAPH_API ngraph::pass::PassBase
     /// Check if this pass has all the pass properties.
     bool get_property(const PassPropertyMask& prop_mask) const;
 
+    void set_name(const std::string& name) { m_name = name; }
+    std::string get_name() const
+    {
+        if (m_name.empty())
+        {
+            const PassBase* p = this;
+            std::string pass_name = typeid(*p).name();
+#ifndef _WIN32
+            int status;
+            pass_name = abi::__cxa_demangle(pass_name.c_str(), nullptr, nullptr, &status);
+#endif
```

Comment:
is it portable? Looks strange

Reply:
at least before this change, it was hidden inside CPP..

Reply:
I'll move it to cpp

Reply:
Done.